src/Chessboard.h

```diff
@@ -3,6 +3,15 @@
 
 #include <boost/shared_ptr.hpp>
 #include <opencv2/core/core.hpp>
+#include <string>
+
+
+class Result{
```

Comment:
Just return a tuple
Use std::tuple. This is not required just to return the ret value and the corners